[ENH] Add the ability of users to specify an explicit HED.xml schema for validation. (#527)
bids-standard/bids-specification: src/99-appendices/03-hed.md

...
```

The direct approach requires that each line in the events file be
annotated. Since there are typically thousands of events in each experiment,
this method of annotation is usually not convenient unless the annotations are
automatically generated. In many experiments, the event instances fall into a
much smaller number of categories, and often these categories are labeled with
numerical codes or short names. It is therefore more convenient to associate
the HED annotations with these categories and allow the analysis tools to make
the association with individual event instances during analysis. To use this
approach, your `*_events.tsv` file should associate a category (often called an
event code) with each event instance. Since BIDS allows an arbitrary number of
columns to be included in an `*_events.tsv` file, you can make this association
by including columns representing various types of event categories in your
`*_events.tsv` file.

Example:

```Text
onset	duration	mycodes
1.1	n/a	Fixation
1.3	n/a	Button
1.8	n/a	Target
...

```
 
The tags in the `HED` column of the `*_events.tsv` file are often specific to the individual event instances,

> while the common properties are represented by categorial values appearing in other columns. 

Reviewer: ```suggestion
while the common properties are represented by categorical values appearing in other columns.
```

Author: Corrected the spelling of categorical.  Not sure about the removing of the space?

       },	  
       "HED": {
           "Fixation": "Event/Category/Experimental stimulus, Event/Label/CrossFix,
		       Event/Description/A cross appears at screen center to serve as a fixation point,
		       Sensory presentation/Visual, Item/Object/2D Shape/Cross,
		       Attribute/Visual/Fixation point, Attribute/Visual/Rendering type/Screen,
		       Attribute/Location/Screen/Center",
           "Target":   "Event/Label/TargetImage, Event/Category/Experimental stimulus,
		       Event/Description/A white airplane as the RSVP target superimposed on a satellite image is displayed.,
		       Item/Object/Vehicle/Aircraft/Airplane, Participant/Effect/Cognitive/Target,
		       Sensory presentation/Visual/Rendering type/Screen/2D),
		       (Item/Natural scene/Arial/Satellite,
		       Sensory presentation/Visual/Rendering type/Screen/2D)",
           "Button":   "Event/Category/Participant response, Event/Label/PressButton, 
		       Event/Description/The participant presses the button as soon as the target is visible,
		       Action/Button press"
        }
   }
}
```
Downstream tools should not distinguish between tags specified using the explicit HED column and 

> the categorical specifications, but should form the union before analysis. Further,
> the normal BIDS inheritance principle applies, so the data dictionaries can
> appear higher in the BIDS hierarchy.  

Reviewer: ```suggestion
the categorical specifications, but should form the union before analysis.
Further, the [inheritance principle](../02-common-principles.md#the-inheritance-principle) applies,
so the data dictionaries can appear higher in the BIDS hierarchy.  
```

Author: Corrected, but again, I am not sure about the rules that should be followed for inserting new lines and the extra spaces after periods.

```Text
onset	duration	mycodes
1.1	n/a	Fixation  

> 1.3	n/a	Button  
> 1.8	n/a	Target  

Reviewer 2: Please remove trailing whitespace (or do not insert new trailing whitespace)

```suggestion
1.3	n/a	Button
1.8	n/a	Target
```

Author: Done sorry....

...

```
 

> The tags in the `HED` column of the \_events.tsv file are often specific to the individual event instances,

Reviewer 2: Please used code formatting with the markdown "backtick" syntax instead of escape characters:
````
\_events.tsv --> `_events.tsv`
````

```suggestion
The tags in the `HED` column of the `*_events.tsv` file are often specific to the individual event instances,
```

Author: Done

while the common properties are represented by categorial values appearing in other columns. 
You may provide a `HED` column and multiple categorical columns to document your events.

> Each of these categorical columns should be documented in a corresponding \_events.json sidecar. 

Reviewer 2: ```suggestion
Each of these categorical columns should be documented in a corresponding `*_events.json` sidecar. 
```

Author: Done

The column name (e.g., `mycodes`) is the dictionary key to this documentation, as illustrated by the following example.  

Example:

```JSON
{ 
   "mycodes": {

>        "LongName": "Local event type names",
> 	   "Description": "Main types of events that comprise a trial",

Reviewer 2: ```suggestion
       "LongName": "Local event type names",
       "Description": "Main types of events that comprise a trial",
```

the common properties associated with categories such as `mycodes` are
encapsulated in the \_events.json dictionary. Downstream tools should not
distinguish between tags specified using the different mechanisms. Further,

> the normal BIDS inheritance principle applies, so these data dictionaries can

Reviewer 2: I suggest to add a link to the BIDS inheritance principle here, and remove the word `"normal"`

```suggestion
the [BIDS inheritance principle](../02-common-principles.md#the-inheritance-principle) applies,
so these data dictionaries can
```

appear higher in the BIDS hierarchy.

The HED vocabulary is specified by a HED schema, which delineates the allowed HED path strings. By default, 
BIDS uses the latest HED schema available in the
[hed-specification](https://github.com/hed-standard/hed-specification/tree/master/hedxml) repository
maintained by the hed-standard group. 

> You can override the default by providing a specific HED version number or including a HED XML file
> in the `sourcedata` directory at the top level of the study. You would indicate the version or
> the XML filename using the optional `HEDVersion` field of the `dataset_description.json` file. 
> The preferred method is to validate with the latest version, but to use the `HEDVersion` field to specify
> which version was used for later reference.  

Reviewer 2: Some wording suggestions:

Please also note that we try to insert a line break after the end of each sentence. --> this "style" makes the future git diff more readable.

```suggestion
Using the latest available HED schema (the default in BIDS) is RECOMMENDED.
However, you can override this default by providing a specific HED version number or by including a HED XML file
in the `sourcedata` directory at the top level of the study.
The HED version number or the path to the HED XML file MUST be specified using the
recommended `HEDVersion` field of the `dataset_description.json` file. 
```
